### DISCUSSION TITLE
Does Axum have HTTPS support?

### DISCUSSION THREAD

**Discussion by kawpii** [tokio-rs/axum#1032] [Q&A] [answered]
I was looking at [this table](https://github.com/flosse/rust-web-framework-comparison#high-level-server-frameworks) and notices there is not HTTPS support in Axum, Is this true?

**Comment by davidpdrsn** [comment, marked as answer]
No that is not true, axum can be run over HTTPS. Check out the [low-level-rustls](https://github.com/tokio-rs/axum/tree/main/examples/low-level-rustls) and [tls-rustls](https://github.com/tokio-rs/axum/tree/main/examples/tls-rustls) examples.

axum is a bit special in that it doesn't come with its own HTTP server (`axum::Server` is just a re-export of `hyper::Server`). Hyper is generic over the underlying transport, ie it can run on any `AsyncRead + AsyncWrite`. That can be a plain HTTP stream or one that is encrypted.

Generally speaking axum is compatible with any server that supports `tower::Service`.
